platform/linuxbsd/wayland/display_server_wayland.cpp

```diff
@@ -680,7 +680,17 @@ void DisplayServerWayland::screen_set_keep_on(bool p_enable) {
 	}
 
 #ifdef DBUS_ENABLED
-	if (screensaver) {
+	if (portal_desktop && portal_desktop->is_inhibit_supported()) {
+		if (p_enable) {
+			WindowID window_id = MAIN_WINDOW_ID;
+			// TODO: Use window IDs for multiwindow support.
```

Comment:
Multiwindow is now supported on Wayland, and focused window can be get using `wayland_thread.pointer_get_pointed_window_id()`. But not sure how much sense it makes to use non-main window for screen server inhibit.